incubator/hnc/pkg/reconcilers/object.go

```diff
@@ -462,18 +456,21 @@ func (r *ObjectReconciler) enqueuePropagatedObjects(ctx context.Context, log log
 
 // operate operates the action generated from syncing the object with the forest.
 func (r *ObjectReconciler) operate(ctx context.Context, log logr.Logger, act syncAction, inst, srcInst *unstructured.Unstructured) error {
+	var err error
+
 	switch act {
 	case actionNop:
-		return nil
+		// nop
 	case actionRemove:
-		return r.deleteObject(ctx, log, inst)
+		err = r.deleteObject(ctx, log, inst)
 	case actionWrite:
-		return r.writeObject(ctx, log, inst, srcInst)
+		err = r.writeObject(ctx, log, inst, srcInst)
 	default:
-		// Generate log for any unset action.
-		log.Error(nil, "ACTION UNSET!!")
-		return nil
+		err = fmt.Errorf("HNC couldn't determine how to update this object (desired action: %s)", act)
```

Comment:
`(desired action: %s)` would be 'actionUnknown`?

Reply:
Yeah, I should make it more clear. This _should_ never happen but if it does, printing an empty string as the reason doesn't really make sense.